src/MICore/LaunchOptions.cs

```diff
@@ -239,9 +231,22 @@ static internal LocalLaunchOptions CreateFromXml(Xml.LaunchOptions.LocalLaunchOp
         public string MIDebuggerServerAddress { get; private set; }
 
         /// <summary>
-        /// [Optional] If supplied, the debugger will attach to the process rather than launching a new one. Note that some operating systems will require admin rights to do this.
+        /// [Required] Path to the executable file. This path must exist on the Visual Studio computer.
         /// </summary>
-        public int ProcessId { get; private set; }
+        public override string ExePath
+        {
+            get
+            {
+                return base.ExePath;
+            }
+            set
+            {
+                if (String.IsNullOrEmpty(value) || !LocalLaunchOptions.CheckPath(value))
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, MICoreResources.Error_InvalidLocalExePath, value));
+
+                base.ExePath = value;
+            }
+        }
```

Comment:
On Fri, Apr 8, 2016 at 6:15 PM -0700, "Gregg Miskelly" <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/MICore/LaunchOptions.cs<https://github.com/Microsoft/MIEngine/pull/259#discussion_r59107215>:

>          /// </summary>
> -        public int ProcessId { get; private set; }
> +        public override string ExePath
> +        {
> +            get
> +            {
> +                return base.ExePath;
> +            }
> +            set
> +            {
> +                if (String.IsNullOrEmpty(value) || !LocalLaunchOptions.CheckPath(value))
> +                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, MICoreResources.Error_InvalidLocalExePath, value));
> +
> +                base.ExePath = value;
> +            }
> +        }


I disabled this code because the file path can be an exe on the PATH. You don't want to bring it back.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/MIEngine/pull/259/files/e954491c978a954faaadea93787112899af6f1c3#r59107215>